rust_src/src/editfns.rs

```diff
@@ -152,3 +153,31 @@ pub fn goto_char(position: LispObject) -> LispObject {
     };
     position
 }
+
+/// Return character in current buffer at position POS.
+/// POS is an integer or a marker and defaults to point.
+/// If POS is out of range, the value is nil.
+#[lisp_fn(min = "0")]
+pub fn char_after(mut pos: LispObject) -> LispObject {
+    let buffer_ref = ThreadState::current_buffer();
+    if pos.is_nil() {
+        pos = point();
+    }
+    if pos.is_marker() {
+        let pos_byte = pos.as_marker().unwrap().bytepos_or_error();
+        // Only works correctly if marker points to current buffer
```

Comment:
I think this comment is misleading: the code works correctly, it's just an unfortunate/surprising behaviour that's implemented.

Could you rephrase this to something like: "Note that this considers the position in the current buffer, even if the marker is from another buffer"?

Reply:
Yeah, I was also thinking about changing this.